cmd/red/test.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"fmt"
+	"github.com/segmentio/conf"
+	"github.com/segmentio/events"
+	eventslog "github.com/segmentio/events/log"
+	redis "github.com/segmentio/redis-go"
+	"github.com/segmentio/stats"
+	"github.com/segmentio/stats/datadog"
+	"math/rand"
+	"os"
+	"time"
+)
+
+type testConfig struct {
+	Proxy     string `conf:"proxy" help:"Address on which the RED proxy under test is listening for incoming connections, in ip:port format." validate:"nonzero"`
+	Dogstatsd string `conf:"dogstatsd" help:"Address of the dogstatsd agent to send metrics to, in ip:port format."                validate:"nonzero"`
+	Batch     int    `conf:"batch" help:"Test batch size: number of keys to read and write in each test run."`
+	Runs      int    `conf:"runs" help:"Number of times to cycle the test. Specify zero to run indefinitely."`
+}
+
+func test(args []string) (err error) {
+	logger := eventslog.NewLogger("", 0, events.DefaultHandler)
+	engine := stats.DefaultEngine
+
+	config := testConfig{Batch: 160}
+
+	conf.LoadWith(&config, conf.Loader{
+		Name: "red proxy",
+		Args: args,
+		Sources: []conf.Source{
+			conf.NewEnvSource("RED", os.Environ()...),
+		},
+	})
+
+	if len(config.Dogstatsd) != 0 {
+		stats.Register(datadog.NewClient(config.Dogstatsd))
+	}
+	defer stats.Flush()
+
+	for i := 0; config.Runs == 0 || i < config.Runs; i++ {
+		logger.Printf("Starting run #%d.", i)
+		transport := &redis.Transport{}
```

Comment:
You can decorate the transport with the `redisstats.NewTransport(redis.RoundTripper) redis.RoundTripper` function, so your client will magically start producing more metrics ;)